addons/crm/models/crm_lead.py

```diff
@@ -388,6 +392,34 @@ def _compute_day_close(self):
             date_close = fields.Datetime.from_string(lead.date_closed)
             lead.day_close = abs((date_close - date_create).days)
 
+    @api.depends('message_ids', 'write_date', 'stage_id.day_rot')
+    def _compute_date_rot(self):
+        for lead in self:
+            # should only fetch the first message with types Email Outgoing or Comment (or Notification, for completed activities)
+            last_message = next(
+                (
+                    message for message in lead.message_ids if message.message_type in ['email_outgoing', 'comment', 'notification']
+                ), False
+            )
+            if last_message:
+                last_activity = max(last_message.date, lead.write_date).date()
```

Comment:
Same comment as above about using _read_group / create_date / activity subtype (in case code moves again :p)